tests/test_modules/test_contact.py

```diff
@@ -0,0 +1,226 @@
+            scwc.add_contact_to(mock_rod, mock_rod_registered)
+        assert "was not found, did you" in str(excinfo.value)
+
+        # Switch arguments
+        with pytest.raises(ValueError) as excinfo:
+            scwc.add_contact_to(mock_rod_registered, mock_rod)
+        assert "was not found, did you" in str(excinfo.value)
+
+    def test_contact_with_illegal_system_throws(self, load_system_with_contacts):
+        scwc = load_system_with_contacts
+
+        # Register this rod
+        mock_rod_registered = self.MockRod(5, 5, 5, 5)
+        scwc.append(mock_rod_registered)
+
+        # Not a rod, but a list!
+        mock_rod = [1, 2, 3, 5]
+
+        with pytest.raises(TypeError) as excinfo:
+            scwc.add_contact_to(mock_rod, mock_rod_registered)
+        assert "not a sys" in str(excinfo.value)
+
+        # Switch arguments
+        with pytest.raises(TypeError) as excinfo:
+            scwc.add_contact_to(mock_rod_registered, mock_rod)
+        assert "not a sys" in str(excinfo.value)
+
+    """
+    END of testing BaseSystem calls
+    """
+
+    def test_contact_registers_and_returns_Contact(self, load_system_with_contacts):
+        scwc = load_system_with_contacts
+
+        mock_rod_one = self.MockRod(2, 3, 4, 5)
+        scwc.append(mock_rod_one)
+
+        mock_rod_two = self.MockRod(4, 5)
+        scwc.append(mock_rod_two)
+
+        _mock_contact = scwc.add_contact_to(mock_rod_one, mock_rod_two)
+        assert _mock_contact in scwc._contacts
+        assert _mock_contact.__class__ == _Contact
+
+    from elastica.contact_forces import NoContact
+
+    @pytest.fixture
+    def load_rod_with_contacts(self, load_system_with_contacts):
+        scwc = load_system_with_contacts
+
+        mock_rod_one = self.MockRod(2, 3, 4, 5)
+        scwc.append(mock_rod_one)
+        mock_rod_two = self.MockRod(5.0, 5.0)
+        scwc.append(mock_rod_two)
+
+        def mock_init(self, *args, **kwargs):
+            pass
+
+        # in place class
+        MockContact = type(
+            "MockContact", (self.NoContact, object), {"__init__": mock_init}
+        )
+
+        # Constrain any and all systems
+        scwc.add_contact_to(0, 1).using(MockContact)  # index based contact
+        scwc.add_contact_to(mock_rod_one, mock_rod_two).using(
+            MockContact
+        )  # system based contact
+        scwc.add_contact_to(0, mock_rod_one).using(
+            MockContact
+        )  # index/system based contact
+
+        return scwc, MockContact
+
+    def test_contact_finalize_correctness(self, load_rod_with_contacts):
+        scwc, contact_cls = load_rod_with_contacts
+
+        scwc._finalize_contact()
+
+        for (fidx, sidx, contact) in scwc._contacts:
+            assert type(fidx) is int
+            assert type(sidx) is int
+            assert type(contact) is contact_cls
+
+    def test_contact_call_on_systems(self):
+        # TODO Finish after the architecture is complete
+        pass
```

Comment:
Remove it or write tests, Architecture was done long time ago :)

Reply:
should I do that for the test_connections too?

Reply:
Sure that will be good

Reply:
added test for test_connections, will add similar test to test_contacts once contact class other than NoContact are added.